src/scan/postgres_table_reader.cpp

```diff
@@ -278,13 +288,37 @@ PostgresTableReader::GetNextTuple() {
 	return PostgresFunctionGuard(ExecClearTuple, slot);
 }
 
+/*
+ * Reads the next minimal tuple from a Postgres parallel worker and copies it into the provided buffer.
+ * This function should only be called when the table scan is running with parallel workers.
+ *
+ * @param minimal_tuple_buffer Buffer to store the copied minimal tuple.
+ * @return true if a tuple was read and copied; false if the scan is complete and no more tuples are available.
+ *
+ * Note: The caller must hold the GlobalProcessLock before invoking this function.
+ */
+bool
+PostgresTableReader::GetNextMinimalWorkerTuple(std::vector<uint8_t> &minimal_tuple_buffer) {
+	MinimalTuple worker_minmal_tuple = GetNextWorkerTuple();
+	if (HeapTupleIsValid(worker_minmal_tuple)) {
+		// deep copy worker_minmal_tuple to destination buffer
+		Size tuple_size = worker_minmal_tuple->t_len + MINIMAL_TUPLE_DATA_OFFSET;
+		minimal_tuple_buffer.resize(tuple_size);
+		memcpy(minimal_tuple_buffer.data(), worker_minmal_tuple, tuple_size);
+		return true;
+	}
+
+	minimal_tuple_buffer.resize(0);
+	return false;
+}
+
 MinimalTuple
 PostgresTableReader::GetNextWorkerTuple() {
 	int nvisited = 0;
 	TupleQueueReader *reader = NULL;
 	MinimalTuple minimal_tuple = NULL;
 	bool readerdone = false;
-	for (;;) {
+	for (; next_parallel_reader < nreaders;) {
```

Comment:
Why is this suddenly needed?

Reply:
In the multi-threaded scenario, one thread might read all worker tuples, perform cleanup, and then release the global lock. When other threads call this function afterward, they will attempt to index an empty array, potentially causing a segmentation fault.